File: server/models/pokemon.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Pokemon struct {
	ID              primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Name            string             `bson:"name" json:"name"`
	Number          int                `bson:"number" json:"number"`
	ImageURL        string             `bson:"imageUrl" json:"imageUrl"`
	Description     string             `bson:"description" json:"description"`
	BackgroundColor string             `bson:"backgroundColor" json:"backgroundColor"`
}

File: pokedex/lib/models/pokemon.dart
import 'package:flutter/material.dart';

class Pokemon {
  final String id;
  final String name;
  final int number;
  final String imageUrl;
  final String description;
  final Color backgroundColor;

  Pokemon({
    required this.id,
    required this.name,
    required this.number,
    required this.imageUrl,
    required this.description,
    required this.backgroundColor,
  });

  factory Pokemon.fromJson(Map<String, dynamic> json) {
    return Pokemon(
      id: json['id'],
      name: json['name'],
      number: json['number'],
      imageUrl: json['imageUrl'],
      description: json['description'],
      backgroundColor: _hexToColor(json['backgroundColor']),
    );
  }

  static Color _hexToColor(String hex) {
    hex = hex.replaceAll('#', '');
    return Color(int.parse('FF$hex', radix: 16));
  }
}
